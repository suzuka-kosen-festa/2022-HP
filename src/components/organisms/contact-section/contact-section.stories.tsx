import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import type { ComponentPropsWithoutRef } from "react";
import { ContactSection } from ".";

type T = typeof ContactSection;
type Story = ComponentStoryObj<T>;
type Meta = ComponentMeta<T>;

const args: ComponentPropsWithoutRef<T> = {
  buttonText: "お問い合わせ",
  children: "第57回鈴鹿高専祭に関するお問い合わせは下のボタンからお願いいたします。(google formsが開きます)",
  link: "#",
  title: "Contact us",
};

export default {
  args,
  argTypes: {
    buttonText: {
      description: "The text for button in section",
      control: {
        type: "text",
      },
    },
    children: {
      description: "Value of main sentence in section",
      control: {
        type: "text",
      },
    },
    link: {
      description: "Href to GoogleForms",
      control: {
        type: "text",
      },
    },
    title: {
      description: "Title of section",
      control: {
        type: "text",
      },
    },
  },
  component: ContactSection,
} as Meta;

export const Default: Story = {};
