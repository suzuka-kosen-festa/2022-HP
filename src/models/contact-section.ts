import type { ComponentProps, ReactHTML } from "react";

export type ContactSectionProperties = Omit<ComponentProps<ReactHTML["section"]>, "children"> & {
  buttonText: string;
  title: string;
  link: string;
  children: string;
};
