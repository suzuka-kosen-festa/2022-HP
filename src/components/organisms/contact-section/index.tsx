import React from "react";
import tw from "twin.macro";
import { ContactSectionProperties } from "../../../models";
import { Button } from "../../atoms/button";
import { Heading } from "../../atoms/heading";
import { Text } from "../../atoms/text";

const ContactSectionContainer = tw.section`space-y-[calc(200vw / 63)]`;

const ContactSection: React.FC<ContactSectionProperties> = ({ buttonText, title, link, children, ...rest }) => (
  <ContactSectionContainer aria-label={"お問い合わせ"} {...rest}>
    <Heading>{title}</Heading>
    <Text>{children}</Text>
    <Button as={"a"} href={link} target={"_blank"} rel={"noopener noreferrer"} role={"button"}>
      {buttonText}
    </Button>
  </ContactSectionContainer>
);

export { ContactSection };
