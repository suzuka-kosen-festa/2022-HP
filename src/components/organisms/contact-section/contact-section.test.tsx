import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./contact-section.stories";

const { Default } = composeStories(stories);

const regionoptions = {
  name: "お問い合わせ",
};

const buttonoptions = {
  name: "お問い合わせ",
};

describe("(components) organisms/contact-section", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=region]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("region", regionoptions)).toBeInTheDocument();
  });

  test("to be [role=button]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("button", buttonoptions)).toBeInTheDocument();
  });

  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
