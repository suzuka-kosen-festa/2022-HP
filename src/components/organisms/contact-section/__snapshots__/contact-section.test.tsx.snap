// Vitest Snapshot v1

exports[`(components) organisms/contact-section > take snap shot 1`] = `
<div>
  <section
    aria-label="お問い合わせ"
    class="css-wkpw2c"
  >
    <h1
      class="css-1461gzd-ContactSection"
    >
      Contact us
    </h1>
    <p
      class="css-25tbau-ContactSection"
    >
      第57回鈴鹿高専祭に関するお問い合わせは下のボタンからお願いいたします。(google formsが開きます)
    </p>
    <a
      class="css-jabq07-Button"
      href="#"
      rel="noopener noreferrer"
      role="button"
      tabindex="0"
      target="_blank"
    >
      <p
        class="css-7bzdgk-Button"
      >
        お問い合わせ
      </p>
    </a>
  </section>
</div>
`;
